File: src/main/java/org/digma/intellij/plugin/insights/DownloadHandlerAdapter.java
package org.digma.intellij.plugin.insights;

import org.cef.browser.CefBrowser;
import org.cef.callback.CefBeforeDownloadCallback;
import org.cef.callback.CefDownloadItem;
import org.cef.callback.CefDownloadItemCallback;
import org.cef.handler.CefDownloadHandler;

class DownloadHandlerAdapter implements CefDownloadHandler {
    @Override
    public void onBeforeDownload(CefBrowser browser, CefDownloadItem downloadItem, String suggestedName, CefBeforeDownloadCallback callback) {
        callback.Continue(suggestedName, true);
    }

    @Override
    public void onDownloadUpdated(CefBrowser browser, CefDownloadItem downloadItem, CefDownloadItemCallback callback) {

    }
}

File: src/main/kotlin/org/digma/intellij/plugin/ui/jcef/DownloadHandlerAdapter.kt
package org.digma.intellij.plugin.ui.jcef

import org.cef.browser.CefBrowser
import org.cef.callback.CefBeforeDownloadCallback
import org.cef.callback.CefDownloadItem
import org.cef.callback.CefDownloadItemCallback
import org.cef.handler.CefDownloadHandler

class DownloadHandlerAdapter : CefDownloadHandler {
    override fun onBeforeDownload(browser: CefBrowser?, dowloadItem: CefDownloadItem?, suggestedName: String?, callback: CefBeforeDownloadCallback?) {
        callback?.Continue(suggestedName, true)
    }

    override fun onDownloadUpdated(browser: CefBrowser?, downloadItem: CefDownloadItem?, callback: CefDownloadItemCallback?) {
    }
}